crates/lib-dialects/test/fixtures/dialects/tsql/table_variables.yml

```diff
@@ -0,0 +1,60 @@
+file:
+- statement:
+  - select_statement:
+    - select_clause:
+      - keyword: SELECT
+      - select_clause_element:
+        - column_reference:
+          - naked_identifier: value
+    - from_clause:
+      - keyword: FROM
+      - from_expression:
+        - from_expression_element:
+          - table_expression:
+            - table_reference:
+              - table_reference:
+                - tsql_variable: '@DepartmentCodes'
+- statement_terminator: ;
+- statement:
+  - select_statement:
+    - select_clause:
+      - keyword: SELECT
+      - select_clause_element:
+        - column_reference:
+          - naked_identifier: ids
+          - dot: .
+          - naked_identifier: value
+    - from_clause:
+      - keyword: FROM
+      - from_expression:
+        - from_expression_element:
+          - table_expression:
+            - table_reference:
+              - table_reference:
+                - tsql_variable: '@DepartmentCodes'
+          - alias_expression:
+            - keyword: AS
+            - naked_identifier: ids
+- statement_terminator: ;
+- statement:
+  - select_statement:
+    - select_clause:
+      - keyword: SELECT
+      - select_clause_element:
+        - wildcard_expression:
+          - wildcard_identifier:
+            - star: '*'
+    - from_clause:
+      - keyword: FROM
+      - from_expression:
+        - from_expression_element:
+          - table_expression:
+            - table_reference:
+              - table_reference:
+                - tsql_variable: '@TableVariable'
+          - keyword: WITH
+          - bracketed:
+            - start_bracket: (
+            - keyword: NOLOCK
+            - end_bracket: )
```

Comment:
nit: I wonder if we should wrap these in a `Table Hint` statement type. 

https://learn.microsoft.com/en-us/sql/t-sql/queries/hints-transact-sql-table?view=sql-server-ver17

Reply:
Yes definitely, looks like I missed that.